Highlight moves on hover in RecordView and BookView

blitzingeagle opened the pull request:
# 説明 / Description

Adds highlight color on moves when hovering over moves list. I think it feels easier to use/prevents misclicks. Let me know what you think! 🙂 

# チェックリスト / Checklist

- MUST
  - [X] `npm test` passed
  - [X] `npm run lint` was applied without warnings
  - [X] changes of `/docs/webapp` not included (except release branch)
  - [X] `console.log` not included (except script file)
- MUST for Outside Contributor
  - [X] understand [プロジェクトへの関わり方について](https://github.com/sunfish-shogi/shogihome/wiki/%E3%83%97%E3%83%AD%E3%82%B8%E3%82%A7%E3%82%AF%E3%83%88%E3%81%B8%E3%81%AE%E9%96%A2%E3%82%8F%E3%82%8A%E6%96%B9%E3%81%AB%E3%81%A4%E3%81%84%E3%81%A6)
- RECOMMENDED (it depends on what you change)
  - [ ] unit test added/updated
  - [ ] i18n

sunfish-shogi commented:
@blitzingeagle 
It's a very interesting idea!

I agree with highlighting the hovered move.
However, I have three concerns.

### 1. strong highlight color

<img width="304" alt="スクリーンショット 2025-03-08 2 12 21" src="https://github.com/user-attachments/assets/a5d5177d-59fc-4399-a434-9d1f636fbc71" />
<img width="304" alt="スクリーンショット 2025-03-08 2 12 31" src="https://github.com/user-attachments/assets/ee16035e-9f28-4388-8cdd-2dcc8f3656a1" />
<img width="304" alt="スクリーンショット 2025-03-08 2 12 42" src="https://github.com/user-attachments/assets/bbfc33d1-db73-45cd-b873-1c1ec2704f08" />
<img width="304" alt="スクリーンショット 2025-03-08 2 13 21" src="https://github.com/user-attachments/assets/47f4f141-72ee-4cfa-ad2b-a36e3b8e2eec" />

I think highlight color should be close to the background color while still maintaining good visibility.

For example, in the following screenshot, you can see how Chrome highlights my bookmark list.

<img width="306" alt="スクリーンショット 2025-03-08 2 28 25" src="https://github.com/user-attachments/assets/aa9ab153-6845-4e08-87c2-03c5fcba00fe" />

### 2. mixed color

Since clicking the selected move does nothing, the mixed color might be unnecessary.

### 3. book moves

The highlighted areas of book moves are not clickable.
Additionally the buttons are already highlighted.

<img width="608" alt="スクリーンショット 2025-03-08 2 10 13" src="https://github.com/user-attachments/assets/0e7109a8-c49d-4be9-9d76-cdf1566ffb6d" />

blitzingeagle commented:
@sunfish-shogi 
Good points. I thought about the same. Here was my thinking.

1. **Strong highlight colors**: I just copied the `--hovered-text-bg-color` from the `--text-bg-color-disabled` colors for each theme, thinking it would best fit the theme. We could change the color values or make it into a better mix if the color is too striking. I just couldn't think of good colors by myself. 😅
2. **Mixed color**: I added mixed color primarily to differentiate a move with branches using `--text-bg-color-warning` when hovered. But afterwards it felt a little weird to hover over a selected move and not see a color difference when all others did. I can make the color change more subtle (right now I made it a 70% original color mix). While there's no clickable functionality for an already selected move I thought the UI's continuity made a bit more sense.
3. **Book moves**: A way to hover and preview moves would be nice, but I haven't figured out the emit function yet. (It's easier for users to see an arrow instead of reading the notation, example here: https://lichess.org/analysis). Perhaps this UI change only makes sense after that gets implemented. I could stash these changes for a future PR. I just enjoyed seeing the reactivity when using the interface. 🙂 

My changes are a reflection of some of my personal preferences as a user of this software, so I hope to get your opinions before anything gets merged!

blitzingeagle commented:
Just adjusted the hovered background color, using 80% color mix for regular moves, branch moves, and selected moves. I think this addresses your concerns about the strong highlight color and mixed color, while preserving the continuity in the UI. Let me know what you think!

blitzingeagle commented:
<img width="200" alt="Screenshot 2025-03-07 at 7 28 30 PM" src="https://github.com/user-attachments/assets/d0ce9300-522a-4106-a298-5d944643b8d8" />
<img width="200" alt="Screenshot 2025-03-07 at 7 28 44 PM" src="https://github.com/user-attachments/assets/66f1a486-06df-444a-a565-be8d5d6fa205" />
<img width="200" alt="Screenshot 2025-03-07 at 7 28 56 PM" src="https://github.com/user-attachments/assets/faf28ce4-83ff-47de-bf92-b964a329a618" />
<img width="200" alt="Screenshot 2025-03-07 at 7 29 07 PM" src="https://github.com/user-attachments/assets/ece7b6d2-7216-4f8e-98cb-7805be4bf569" />
<img width="200" alt="Screenshot 2025-03-07 at 7 29 14 PM" src="https://github.com/user-attachments/assets/37db7e2a-35fb-4f0b-8334-53a63fd1f881" />
<img width="200" alt="Screenshot 2025-03-07 at 7 29 21 PM" src="https://github.com/user-attachments/assets/67e73b9a-2ae0-4f83-a2fe-5af9289906ca" />

sunfish-shogi commented:
### color code

The subtle colors above look better than the first version.
However, these colors generated by `color-mix` function appear cloudy because the chroma or hue values changed.
I think hovered-text-bg-color should maintain the same chroma and hue as the text-background-color.

I used to pick color codes by [0to255](https://0to255.com/), but this service is now paid-only.
For example, you can use [Adobe Color](https://color.adobe.com/) instead.

<img width="300" alt="スクリーンショット 2025-03-08 15 40 29" src="https://github.com/user-attachments/assets/fac89366-1e7f-4b94-af0f-2d1968f10818" />

<img width="252" alt="スクリーンショット 2025-03-08 15 43 50" src="https://github.com/user-attachments/assets/17db309d-963e-419c-aa6f-c9953aea6bde" />
<img width="293" alt="スクリーンショット 2025-03-08 15 41 35" src="https://github.com/user-attachments/assets/b0203e54-d440-43e9-aa39-e694c9ac949c" />

### selected move and book moves

In this app, the highlighting is only used to indicate interactive elements.
For example, the disabled buttions are not highlighted.

Many game apps use a variety visual and audio effects.
However, this app primarily targets researchers and developers.
I believe such effects should be applied only where necessary.

Additionaly, features based on mouse hovering provide no benefit to touch device users.
I think highlights for auxiliary purposes are useful, but it should not be the primary means of UI interaction.

blitzingeagle commented:
@sunfish-shogi Understood. Changed the color-mix method to HSL, and removed highlighting for non-clickable components. I still think hover (or tap for mobile) of book moves should display a corresponding arrow on the board, and the corresponding book move should be highlighted and be beneficial to researchers and developers. But that can belong in a separate PR.

sunfish-shogi commented:
@blitzingeagle 
Sorry, my previous comment meant not to use color-mix.
I always adjust color codes individually.

Also, could you avoid making changes to BookView.vue?
Please do not modify its layout.
